test/controllers/EmailControllerSpec.scala

```diff
@@ -45,6 +46,26 @@ class EmailControllerSpec extends SpecBase {
         status(result) shouldBe OK
       }
     }
+
```

Comment:
Could you please refactor this class as below cause we are duplicating the things as of now? Its good to have mocks in individual tests.

class EmailControllerSpec extends SpecBase {

  "EmailController" must {
    "return unverified email" in new Setup {
      when[Future[EmailUnverifiedResponse]](
        mockHttpClient.GET(ArgumentMatchers.endsWith("/subscriptions/unverified-email-display"), any, any)
        (any, any, any))
        .thenReturn(Future.successful(response))

      running(app) {
        val connector = app.injector.instanceOf[CustomsFinancialsApiConnector]

        val result: Future[Option[String]] = connector.isEmailUnverified(hc)
        await(result) mustBe expectedResult
      }
    }

    "return unverified email response" in new Setup {
      when[Future[EmailUnverifiedResponse]](
        mockHttpClient.GET(ArgumentMatchers.endsWith("/subscriptions/unverified-email-display"), any, any)
        (any, any, any))
        .thenReturn(Future.successful(response))

      running(app) {
        val request = fakeRequest(GET, routes.EmailController.showUnverified().url)
        val result = route(app, request).value
        status(result) shouldBe OK
      }
    }

    "return undeliverable email response" in new Setup {

      when[Future[EmailVerifiedResponse]](
        mockHttpClient.GET(ArgumentMatchers.endsWith("/subscriptions/email-display"), any, any)
        (any, any, any))
        .thenReturn(Future.successful(EmailVerifiedResponse(Some("undeliverableEmail"))))

      running(app) {
        val request = fakeRequest(GET, routes.EmailController.showUndeliverable().url)
        val result = route(app, request).value
        status(result) shouldBe OK
      }
    }
  }

  trait Setup {
    val expectedResult = Some("unverifiedEmail")
    implicit val hc: HeaderCarrier = HeaderCarrier()
    val mockHttpClient = mock[HttpClient]

    val response = EmailUnverifiedResponse(Some("unverifiedEmail"))

    val app = applicationBuilder().overrides(
      bind[HttpClient].toInstance(mockHttpClient)
    ).build()
  }

}